compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Conversions.scala

```diff
@@ -88,58 +89,78 @@ final class Conversions(
+            builder.setUnhandledException(convertValue(value))
+          case UserError(msg) =>
+            builder.setUserError(msg)
+          case ContractNotFound(cid) =>
+            // TODO https://github.com/digital-asset/daml/issues/9974
+            //   we should probably use ${cid.coid} instead of $cid
+            setCrash(s"contract $cid not found")
+          case TemplatePreconditionViolated(tid, optLoc, arg) =>
+            val uepvBuilder = proto.ScenarioError.TemplatePreconditionViolated.newBuilder
+            optLoc.map(convertLocation).foreach(uepvBuilder.setLocation)
+            builder.setTemplatePrecondViolated(
+              uepvBuilder
+                .setTemplateId(convertIdentifier(tid))
+                .setArg(convertValue(arg))
+                .build
+            )
+          case LocalContractNotActive(coid, tid, consumedBy) =>
+            builder.setUpdateLocalContractNotActive(
+              proto.ScenarioError.ContractNotActive.newBuilder
+                .setContractRef(mkContractRef(coid, tid))
+                .setConsumedBy(proto.NodeId.newBuilder.setId(consumedBy.toString).build)
+                .build
+            )
+          case LocalContractKeyNotVisible(coid, gk, actAs, readAs, stakeholders) =>
+            builder.setScenarioContractKeyNotVisible(
+              proto.ScenarioError.ContractKeyNotVisible.newBuilder
+                .setContractRef(mkContractRef(coid, gk.templateId))
+                .addAllActAs(actAs.map(convertParty(_)).asJava)
+                .addAllReadAs(readAs.map(convertParty(_)).asJava)
+                .addAllStakeholders(stakeholders.map(convertParty).asJava)
+                .build
+            )
+          case ContractKeyNotFound(gk) =>
+            builder.setScenarioContractKeyNotFound(
+              proto.ScenarioError.ContractKeyNotFound.newBuilder
+                .setTemplateId(convertIdentifier(gk.templateId))
+                .setKey(convertValue(gk.key))
+                .build
+            )
+          case DuplicateContractKey(key) =>
+            builder.setScenarioCommitError(
+              proto.CommitError.newBuilder.setUniqueKeyViolation(convertGlobalKey(key)).build
+            )
+          case CreateEmptyContractKeyMaintainers(tid, arg, key) =>
+            builder.setCreateEmptyContractKeyMaintainers(
+              proto.ScenarioError.CreateEmptyContractKeyMaintainers.newBuilder
+                .setArg(convertValue(arg))
+                .setTemplateId(convertIdentifier(tid))
+                .setKey(convertValue(key))
+            )
+          case FetchEmptyContractKeyMaintainers(tid, key) =>
+            builder.setFetchEmptyContractKeyMaintainers(
+              proto.ScenarioError.FetchEmptyContractKeyMaintainers.newBuilder
+                .setTemplateId(convertIdentifier(tid))
+                .setKey(convertValue(key))
+            )
+          case wtc: WronglyTypedContract =>
+            sys.error(
```

Comment:
this isn’t actually true. you can easily `coerceContractId` your way into this error. Not for this PR though.

Reply:
https://gist.github.com/cocreature/a9a2f55031413f50c7eef5d5bda7cb17 should do the trick